test/jdk/javax/swing/JTabbedPane/TestJTabbedPaneOpaqueColor.java

```diff
@@ -0,0 +1,236 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.GridLayout;
+import javax.swing.AbstractAction;
+import javax.swing.AbstractButton;
+import javax.swing.JButton;
+import javax.swing.JCheckBox;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTabbedPane;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+/*
+ * @test
+ * @bug 8226990
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary Test JTabbedPane's contentarea, tab area and tab color
+ *          for different LAFs when opacity is enabled or disabled.
+ * @run main/manual TestJTabbedPaneOpaqueColor
+ */
+
+public class TestJTabbedPaneOpaqueColor {
+    private static JFrame frame;
+    private static JTabbedPane tabPane;
+    private static final String INSTRUCTIONS = """
+            The background color of panel (which contains the tabbed pane is green).
+            The background color of the tabbed pane is red.
+            The TabbedPane is not opaque initially.
+            For 'Content Opaque' and 'Tabs Opaque' to have effect, tab pane opacity should
+            be set to false i.e. Opaque checkbox should be unchecked.
+
+            Check the default behaviour of the tabbed pane:
+              - the area behind tabs is transparent (it must be green).
+              - the tabs area is opaque (it must be red, except the selected tab which must be gray).
+              - the content area is opaque (it must be gray).
+
+            Test Case 1 (Test TabPane Opaque behaviour):
+
+            Click to the check box 'opaque'. And be sure that
+              - the area behind tabs is opaque (it must be red).
+              - the tabs area is opaque (it must be red, except the selected tab which must be gray).
+              - the content area is opaque (it must be gray).
+
+            Check this behaviour for other LAFs and tab layout.
+            NOTE: For Nimbus LAF, tabs color are specific to nimbus style.
+
+            Test Case 2 (Test Content Opaque UI property):
+            To enable Content Opaque checkbox, make Sure "Opaque checkbox" is not selected.
+
+            Click to the check box 'content opaque' and be sure that
+              - the content area is transparent (it must be green).
+              - Select the 'content opaque' and check that content area is opaque
+                (it must be gray).
+            Check this behaviour for other LAFs and tab layout.
```

Comment:
Click to check box 'content opaque' might seem unclear, whether the checkbox needs to be checked or unchecked. It is better to replace it with phrases - "checked/unchecked" or "select/unselect" as below.

Same applies to Case 1 and Case 3.

```
Test Case 2 - Test Content pane opacity:
To test Content pane opacity, make sure "Opaque checkbox" is UNCHECKED.

Verify the following with 'content opaque' option:
- when checked: the content area should be opaque (it must be gray).
- when unchecked: the content area should be transparent (it must be green).
           
Check this behavior for other LAFs and tab layout.
```

Reply:
This looks better than before. Updated the instruction.